engine/src/distribution/primitives.cpp

```diff
@@ -263,10 +269,16 @@ void distributePartitions(Context * context, std::vector<NodeColumnView> & parti
 		}
 		BlazingTableView columns = nodeColumn.second;
 		auto destination_node = nodeColumn.first;
-		threads.push(([message_id, context_token, self_node, destination_node, columns](int thread_id) mutable {
+		auto send_function = threads.push(([message_id, context_token, self_node, destination_node, columns](int thread_id) mutable {
```

Comment:
this wont work because basically every kernel jsut became single threaded with this use case.